### ISSUE TITLE
Cannot install matplotlib in Tensorflow environment

### ISSUE THREAD

**Issue by rueberger** [conda/conda#2783]
I'm unable to install matplotlib to do an issue with numpy in my tensorflow environment.

Output of `conda info`:

```
Current conda install:

             platform : osx-64
        conda version : 4.1.0
  conda-build version : 0+unknown
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : /Users/andrew/anaconda  (writable)
  default environment : /Users/andrew/anaconda/envs/tensorflow
     envs directories : /Users/andrew/anaconda/envs
        package cache : /Users/andrew/anaconda/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/osx-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/osx-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False
```

### Steps to reproduce
Following the Tensorflow installation guide directly 
1. `conda create -n tensorflow python=3.5 `
```
Fetching package metadata .......
Solving package specifications .............

Package plan for installation in environment /Users/andrew/anaconda/envs/tensorflow:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    setuptools-23.0.0          |           py35_0         460 KB

The following NEW packages will be INSTALLED:

    openssl:    1.0.2h-1     
    pip:        8.1.2-py35_0 
    python:     3.5.1-0      
    readline:   6.2-2        
    setuptools: 23.0.0-py35_0
    sqlite:     3.13.0-0     
    tk:         8.5.18-0     
    wheel:      0.29.0-py35_0
    xz:         5.0.5-1      
    zlib:       1.2.8-3      

Proceed ([y]/n)? y

Fetching packages ...
setuptools-23. 100% |##############################################################################################################################################################################################| Time: 0:00:00   5.23 MB/s
Extracting packages ...
[      COMPLETE      ]|#################################################################################################################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ]|#################################################################################################################################################################################################################| 100%

```
2. `source activate tensorflow`
3. `export TF_BINARY_URL=https://storage.googleapis.com/tensorflow/mac/tensorflow-0.9.0rc0-py3-none-any.whl`
4. sudo pip install --upgrade $TF_BINARY_URL
```
The directory '/Users/andrew/Library/Caches/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/Users/andrew/Library/Caches/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting tensorflow==0.9.0rc0 from https://storage.googleapis.com/tensorflow/mac/tensorflow-0.9.0rc0-py3-none-any.whl
  Downloading https://storage.googleapis.com/tensorflow/mac/tensorflow-0.9.0rc0-py3-none-any.whl (25.8MB)
    100% |████████████████████████████████| 25.8MB 26kB/s 
Collecting protobuf==3.0.0b2 (from tensorflow==0.9.0rc0)
  Downloading protobuf-3.0.0b2-py2.py3-none-any.whl (326kB)
    100% |████████████████████████████████| 327kB 1.3MB/s 
Requirement already up-to-date: wheel>=0.26 in ./tensorflow/lib/python3.5/site-packages (from tensorflow==0.9.0rc0)
Collecting six>=1.10.0 (from tensorflow==0.9.0rc0)
  Downloading six-1.10.0-py2.py3-none-any.whl
Collecting numpy>=1.10.1 (from tensorflow==0.9.0rc0)
  Downloading numpy-1.11.0-cp35-cp35m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (3.8MB)
    100% |████████████████████████████████| 3.8MB 182kB/s 
Requirement already up-to-date: setuptools in ./tensorflow/lib/python3.5/site-packages/setuptools-23.0.0-py3.5.egg (from protobuf==3.0.0b2->tensorflow==0.9.0rc0)
Installing collected packages: six, protobuf, numpy, tensorflow
Successfully installed numpy-1.11.0 protobuf-3.0.0b2 six-1.10.0 tensorflow-0.9.0rc0
```

5. `conda install ipython` This step is possibly unnecessary to reproduce - Tensorflow now successfully imports 
6. `conda install matplotlib`
```
Fetching package metadata .......
Solving package specifications .............

Package plan for installation in environment /Users/andrew/anaconda/envs/tensorflow:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    qt-4.8.7                   |                3        37.7 MB
    pyqt-4.11.4                |           py35_3         4.0 MB
    ------------------------------------------------------------
                                           Total:        41.7 MB

The following NEW packages will be INSTALLED:

    cycler:          0.10.0-py35_0    
    freetype:        2.5.5-1          
    libpng:          1.6.22-0         
    matplotlib:      1.5.1-np111py35_0
    mkl:             11.3.3-0         
    numpy:           1.11.0-py35_1    
    pyparsing:       2.1.4-py35_0     
    pyqt:            4.11.4-py35_3    
    python-dateutil: 2.5.3-py35_0     
    pytz:            2016.4-py35_0    
    qt:              4.8.7-3          
    sip:             4.16.9-py35_0    
    six:             1.10.0-py35_0    

Proceed ([y]/n)? y

Fetching packages ...
qt-4.8.7-3.tar 100% |##############################################################################################################################################################################################| Time: 0:00:01  39.41 MB/s
pyqt-4.11.4-py 100% |##############################################################################################################################################################################################| Time: 0:00:00   5.48 MB/s
Extracting packages ...
[      COMPLETE      ]|#################################################################################################################################################################################################################| 100%
Linking packages ...
An unexpected error has occurred, please consider sending the##########                                                                                                                                                                 |  23%
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/Users/andrew/anaconda/envs/tensorflow/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/cli/main.py", line 120, in main
    args_func(args, p)
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/cli/main.py", line 127, in args_func
    args.func(args, p)
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/cli/main_install.py", line 69, in execute
    install(args, parser, 'install')
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/cli/install.py", line 407, in install
    execute_actions(actions, index, verbose=not args.quiet)
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/plan.py", line 566, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/instructions.py", line 137, in execute_instructions
    cmd(state, arg)
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/instructions.py", line 80, in LINK_CMD
    link(state['prefix'], dist, lt, index=state['index'], shortcuts=shortcuts)
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/install.py", line 1069, in link
    update_prefix(join(prefix, f), prefix, placeholder, mode)
  File "/Users/andrew/anaconda/lib/python3.5/site-packages/conda/install.py", line 392, in update_prefix
    with open(path, 'rb') as fi:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/andrew/anaconda/envs/tensorflow/lib/python3.5/site-packages/numpy/distutils/site.cfg'
```

Any ideas?

**Comment by kalefranz** [comment]
Try this...

    conda create -n tensorflow python=3.5 matplotlib pip ipython
    source activate tensorflow
    export TF_BINARY_URL=https://storage.googleapis.com/tensorflow/mac/tensorflow-0.9.0rc0-py3-none-any.whl
    pip install --upgrade $TF_BINARY_URL

Let me know how that works.  If it fails, please provide output again.

A couple notes:

* never *ever* use `sudo` with pip unless you know exactly why and what you're doing
* we'll have tensorflow and several other packages in the genre built natively for conda in the coming couple of months

**Comment by kalefranz** [comment]
Hoping that solved the problem. Closing the issue.

**Comment by rueberger** [comment]
Apologies, forgot to follow up - if I recall correctly, the issue was resolved by a fresh install of my tensorflow virtual environment without a clear culprit for the failure. 

Looking forward to native tensorflow builds!

**Comment by ahsannabi** [comment]
I used @kalefranz hints and that worked. My updated tensorflow environment was not picking matplotlib so I deleted the env library in anaconda3. Then the following commands
```
conda create -n tensorflow python=3.5 matplotlib pip ipython
activate tensorflow
pip install --upgrade https://github.com/tensorflow/tensorflow

```Then I restarted my anaconda3 navigator. Changed the application channel from root to tensorflow, changed the spyder status from 'Launch' to 'Install'. I installed spyder again and this time Spyder worked with pip matplotlib ipython and tensorflow code. 

Note: I had to repeat almost the same process for sklearn
1. Close Spyder
2. Activate Tensorflow on Anaconda CommandPrompt
3. Run `conda update scikit-learn` and note the env library in anaconda3 folder
4. Open Anaconda Navigator
5. Choose app channel tensorflow
6. Open Spyder and run code to check for more requirements
